pac4j-http/src/main/java/org/pac4j/http/credentials/authenticator/RestAuthenticator.java

```diff
@@ -0,0 +1,142 @@
+package org.pac4j.http.credentials.authenticator;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.pac4j.core.context.HttpConstants;
+import org.pac4j.core.context.WebContext;
+import org.pac4j.core.credentials.UsernamePasswordCredentials;
+import org.pac4j.core.credentials.authenticator.Authenticator;
+import org.pac4j.core.exception.CredentialsException;
+import org.pac4j.core.exception.HttpAction;
+import org.pac4j.core.exception.TechnicalException;
+import org.pac4j.core.profile.definition.CommonProfileDefinition;
+import org.pac4j.core.profile.definition.ProfileDefinitionAware;
+import org.pac4j.core.util.CommonHelper;
+import org.pac4j.core.util.HttpUtils;
+import org.pac4j.http.profile.RestProfile;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Authenticates against a REST API. The username/password are passed as a basic auth via a POST request,
+ * the JSON response is a user profile.
+ *
+ * @author Jerome Leleu
+ * @since 2.1.0
+ */
+public class RestAuthenticator extends ProfileDefinitionAware<RestProfile> implements Authenticator<UsernamePasswordCredentials> {
+
+    private static final Logger logger = LoggerFactory.getLogger(RestAuthenticator.class);
+
+    private static ObjectMapper mapper;
+
+    static {
+        mapper = new ObjectMapper();
+        mapper.configure(JsonParser.Feature.ALLOW_UNQUOTED_FIELD_NAMES, true);
+        mapper.configure(JsonParser.Feature.ALLOW_SINGLE_QUOTES, true);
+    }
+
+    private String url;
+
+    public RestAuthenticator() {}
+
+    public RestAuthenticator(final String url) {
+        this.url = url;
+    }
+
+    @Override
+    protected void internalInit(final WebContext context) {
+        CommonHelper.assertNotBlank("url", url);
+        defaultProfileDefinition(new CommonProfileDefinition<>(x -> new RestProfile()));
+    }
+
+    @Override
+    public void validate(final UsernamePasswordCredentials credentials, final WebContext context) throws HttpAction, CredentialsException {
+        init(context);
+
+        final String username = credentials.getUsername();
+        final String password = credentials.getPassword();
+        if (CommonHelper.isBlank(username) || CommonHelper.isBlank(password)) {
+            logger.error("Empty username or password");
```

Comment:
should this be an error? if a user provides the wrong data, are we going to log it everytime? for me it is an expected situation, not an error... at most a warning, but I would go for an information or even a debug log: the only reason I would need to see that information is if I'm trying to understand why something is not working as expected, not everytime a user provides wrong data in prod...
at the same time, you could also argue that it is the responsibility of another component of the system to enforce that? your choice, I just wanted to raise the point :)